code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -593,6 +608,22 @@
 			udder.trans_type_to(G, milk_type, rand(5, 10))
 			return
 
+	if(has_toxingland && stat == CONSCIOUS)
+		var/obj/item/reagent_containers/glass/ctcontainer = attacking_item
+		if(istype(src, /mob/living/simple_animal/carp/fluff/ginny) && istype(ctcontainer) && ctcontainer.is_open_container())
```

Comment:
`istype(src, ...)` is not acceptable code. Make this a separate proc and then override it on mobs as needed.